crates/matrix-sdk/src/client/builder.rs

```diff
@@ -129,12 +129,12 @@ impl ClientBuilder {
     /// This is a shorthand for
     /// <code>.[store_config](Self::store_config)([matrix_sdk_sled]::[make_store_config](matrix_sdk_sled::make_store_config)(path, passphrase)?)</code>.
     #[cfg(feature = "sled")]
-    pub fn sled_store(
+    pub async fn sled_store(
```

Comment:
hmm... this is a bit odd of a pattern ... generally, in the builder-pattern we expect to just set things and can chain them and don't have them do stuff yet but only really become active on `build().await?` ...

This isn't strictly about this PR, it just became more obvious that this isn't really proper when I noticed the new `.await` on its setter...